image.add_to_layer() example breaks with newer versions of `image`

The docs claim that this crate depends on [image ^0.23](https://docs.rs/crate/image/%5E0.23) and ships with it but running the code example produces this error:

```
extern crate printpdf;

use printpdf::*;
use std::io::Cursor;
use image_crate::bmp::BmpDecoder;
use std::fs::File;
use std::io::BufWriter;

fn main() {
    let (doc, page1, layer1) = PdfDocument::new("printpdf graphics test", Mm(210.0), Mm(297.0), "Layer 1");
    let current_layer = doc.get_page(page1).get_layer(layer1);

    // currently, the only reliable file formats are bmp/jpeg/png
    // this is an issue of the image library, not a fault of printpdf

    let image_bytes = include_bytes!("../assets/img/BMP_test.bmp");
    let mut reader = Cursor::new(image_bytes.as_ref());

    let decoder = BmpDecoder::new(&mut reader).unwrap();
    let image = Image::try_from(decoder).unwrap();

    let rotation_center_x = Px((image.image.width.0 as f64 / 2.0) as usize);
    let rotation_center_y = Px((image.image.height.0 as f64 / 2.0) as usize);

    // layer,
    image.add_to_layer(current_layer.clone(), ImageTransform {
        rotate: Some(ImageRotation {
            angle_ccw_degrees: 45.0,
            rotation_center_x,
            rotation_center_y,
        }),
        translate_x: Some(Mm(10.0)),
        translate_y: Some(Mm(10.0)),
        .. Default::default()
    });

    doc.save(&mut BufWriter::new(File::create("test_image.pdf").unwrap())).unwrap();
}
```
Error:
```
error[E0433]: failed to resolve: use of undeclared crate or module `image_crate`
 --> src/main.rs:5:5
  |
5 | use image_crate::bmp::BmpDecoder;
  |     ^^^^^^^^^^^ use of undeclared crate or module `image_crate`
```

The image crate no longer has `image::bmp::BmpDecoder`. It has been renamed to `image::codecs::bmp::BmpDecoder`. Okay, let's add the image crate manually and try again:

Cargo.toml:
```
[package]
name = "printpdf"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
printpdf = "0.5.0"
image = "0.23.0"
```
main.rs:
```
extern crate printpdf;

use printpdf::{ImageRotation, ImageTransform, Mm, Px, Image, PdfDocument};
use image::codecs::bmp::BmpDecoder;
use std::io::Cursor;
use std::fs::File;
use std::io::BufWriter;

fn main() {
    let (doc, page1, layer1) = PdfDocument::new("printpdf graphics test", Mm(210.0), Mm(297.0), "Layer 1");
    let current_layer = doc.get_page(page1).get_layer(layer1);

    // currently, the only reliable file formats are bmp/jpeg/png
    // this is an issue of the image library, not a fault of printpdf

    let image_bytes = include_bytes!("../assets/img/BMP_test.bmp");
    let mut reader = Cursor::new(image_bytes.as_ref());

    let decoder = BmpDecoder::new(&mut reader).unwrap();
    let image = Image::try_from(decoder).unwrap();

    let rotation_center_x = Px((image.image.width.0 as f64 / 2.0) as usize);
    let rotation_center_y = Px((image.image.height.0 as f64 / 2.0) as usize);

    // layer,
    image.add_to_layer(current_layer.clone(), ImageTransform {
        rotate: Some(ImageRotation {
            angle_ccw_degrees: 45.0,
            rotation_center_x,
            rotation_center_y,
        }),
        translate_x: Some(Mm(10.0)),
        translate_y: Some(Mm(10.0)),
        .. Default::default()
    });

    doc.save(&mut BufWriter::new(File::create("test_image.pdf").unwrap())).unwrap();
}
```
Error:
```
error[E0277]: the trait bound `printpdf::Image: From<BmpDecoder<&mut std::io::Cursor<_>>>` is not satisfied
  --> src/main.rs:20:17
   |
20 |     let image = Image::try_from(decoder).unwrap();
   |                 ^^^^^^^^^^^^^^^ the trait `From<BmpDecoder<&mut std::io::Cursor<_>>>` is not implemented for `printpdf::Image`
   |
   = help: the trait `From<ImageXObject>` is implemented for `printpdf::Image`
   = note: required because of the requirements on the impl of `Into<printpdf::Image>` for `BmpDecoder<&mut std::io::Cursor<_>>`
   = note: required because of the requirements on the impl of `TryFrom<BmpDecoder<&mut std::io::Cursor<_>>>` for `printpdf::Image`

For more information about this error, try `rustc --explain E0277`.
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -20,7 +20,7 @@ time = { version = "0.2.11", default-features = false, features = ["std"] }
 # optional: logging
 log = { version = "0.4.8", optional = true }
 # image reading (png / jpeg)
-image = { version = "0.23", optional = true, default-features = false, features = ["gif", "jpeg", "png", "pnm", "tiff", "bmp"] }
+image = { version = "0.24.3", optional = true, default-features = false, features = ["gif", "jpeg", "png", "pnm", "tiff", "bmp"] }
 # svg support (svg -> pdf xobject)
 svg2pdf = { version = "0.1.0", optional = true }
 pdf-writer = { version = "0.4.1", optional = true }

--- a/examples/image.rs
+++ b/examples/image.rs
@@ -2,7 +2,7 @@ extern crate printpdf;
 
 use printpdf::*;
 use std::io::Cursor;
-use image_crate::bmp::BmpDecoder;
+use image_crate::codecs::bmp::BmpDecoder;
 use std::fs::File;
 use std::io::BufWriter;
 

--- a/src/color.rs
+++ b/src/color.rs
@@ -74,8 +74,8 @@ impl From<image_crate::ColorType> for ColorSpace {
         match color_type {
             L8 | L16 => ColorSpace::Greyscale,
             La8 | La16 => ColorSpace::GreyscaleAlpha,
-            Rgb8 | Bgr8 | Rgb16 => ColorSpace::Rgb,
-            Rgba8 | Bgra8 | Rgba16 => ColorSpace::Rgba,
+            Rgb8 | Rgb16 => ColorSpace::Rgb,
+            Rgba8 | Rgba16 => ColorSpace::Rgba,
             _ => ColorSpace::Greyscale, // unreachable
         }
     }
@@ -113,7 +113,7 @@ impl From<image_crate::ColorType> for ColorBits {
         use ColorBits::*;
 
         match color_type {
-            L8 | La8 | Rgb8 | Rgba8 | Bgr8 | Bgra8 => Bit8,
+            L8 | La8 | Rgb8 | Rgba8 => Bit8,
             L16 | La16 | Rgb16 | Rgba16 => Bit16,
             _ => Bit8, // unreachable
 

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -136,7 +136,7 @@
 //!     // currently, the only reliable file formats are bmp/jpeg/png
 //!     // this is an issue of the image library, not a fault of printpdf
 //!     let mut image_file = File::open("assets/img/BMP_test.bmp").unwrap();
-//!     let image = Image::try_from(image_crate::bmp::BmpDecoder::new(&mut image_file).unwrap()).unwrap();
+//!     let image = Image::try_from(image_crate::codecs::bmp::BmpDecoder::new(&mut image_file).unwrap()).unwrap();
 //!
 //!     // translate x, translate y, rotate, scale x, scale y
 //!     // by default, an image is optimized to 300 DPI (if scale is None)

